// vocs/_lib/vite/.vocs/dist/assets/benefits-and-risks-CcCPSRyk.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Validator Quickstart",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    h2: "h2",
    header: "header",
    li: "li",
    p: "p",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "validator-quickstart",
        children: ["Validator Quickstart", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#validator-quickstart",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.h2, {
      id: "benefits",
      children: ["Benefits", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#benefits",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Like flashbots & mev-boost, interstate offers a way to earn additional rewards for validators without having to put up any capital and minimal risk. It is almost a free lunch :)."
    }), "\n", jsx(_components.p, {
      children: "We expect validator rewards to be in the 0.2% range with preconfirmation tips and token incentives. We also offer an incentivized testnet program which can offer much more in rewards."
    }), "\n", jsx(_components.p, {
      children: "As interstate develops, you can expect substantially higher rewards for participating in the network."
    }), "\n", jsxs(_components.h2, {
      id: "risks",
      children: ["Risks", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#risks",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: ["Lower MEV-Boost Rewards", "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Validators currently earn ~1.83% from Flashbots / MEV-Boost. The risk of allowing preconfirmations on your validator is that if the preconfirmation network misprices the preconfirmation, space in the block is sold for less than its worth and the block builder has less space to operate with, causing them to create blocks that are less mev-valuable than before."
          }), "\n", jsx(_components.li, {
            children: "Interstate constantly monitors block rewards and ensures the proposer is strictly profitable, removing this constraint"
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["Missed Blocks", "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "We are audited by Spearbit and other leading auditors to ensure code correctness.\nIf there is a bug in interstate code, it could cause you to miss a block, getting penalized by Ethereum network"
          }), "\n", jsx(_components.li, {
            children: "This is unlikely however, Interstate has an insurance fund to compensate the proposer in this case. Please reach out for details. Please note this is not a binding legal agreement unless one has been formally made with you."
          }), "\n"]
        }), "\n"]
      }), "\n", jsxs(_components.li, {
        children: ["VPC Exposure", "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "We are audited by Spearbit and other leading auditors to ensure code correctness."
          }), "\n", jsx(_components.li, {
            children: "A code bug could enable an outside entity to gain control of your vpc."
          }), "\n"]
        }), "\n"]
      }), "\n"]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
