// vocs/_lib/vite/.vocs/dist/assets/faqs-lqH1-GfR.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "FAQs",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    hr: "hr",
    p: "p",
    strong: "strong",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "faqs",
        children: ["FAQs", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#faqs",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.h2, {
      id: "what-is-interstate",
      children: ["What is Interstate?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#what-is-interstate",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Interstate is an all-in-one onchain trading platform built so you never need to switch between apps. We bring together a full suite of integrations, letting you trade the most in-demand assets from a single interface."
    }), "\n", jsxs(_components.h2, {
      id: "is-interstate-a-wallet-too",
      children: ["Is Interstate a wallet too?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#is-interstate-a-wallet-too",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Absolutely — and it goes beyond that. Interstate is a hybrid crypto trading app and wallet rolled into one. We currently support Solana, with additional chains on the way. Your wallet is protected by top-tier security infrastructure, meaning your assets stay under your control at all times — no exceptions."
    }), "\n", jsxs(_components.h2, {
      id: "how-does-interstate-keep-my-funds-secure",
      children: ["How does Interstate keep my funds secure?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-does-interstate-keep-my-funds-secure",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Interstate employs best-in-class security practices including non-custodial architecture, robust key management, and end-to-end encryption. For more details on our wallet infrastructure, check out our partner ", jsx(_components.a, {
        href: "https://turnkey.com",
        children: "Turnkey"
      }), "."]
    }), "\n", jsxs(_components.h2, {
      id: "can-i-purchase-crypto-directly-on-interstate",
      children: ["Can I purchase crypto directly on Interstate?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#can-i-purchase-crypto-directly-on-interstate",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Yes. Through our integration with Coinbase, you can purchase up to $500 in crypto per week without KYC requirements."
    }), "\n", jsxs(_components.h2, {
      id: "can-i-earn-yield-on-my-holdings",
      children: ["Can I earn yield on my holdings?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#can-i-earn-yield-on-my-holdings",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Interstate makes it easy to earn passive income on your assets. We offer up to 15% APY with instant withdrawals available at any time. For more information, visit our partner ", jsx(_components.a, {
        href: "https://marginfi.com",
        children: "Marginfi"
      }), "."]
    }), "\n", jsxs(_components.h2, {
      id: "is-interstate-decentralized",
      children: ["Is Interstate decentralized?", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#is-interstate-decentralized",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Yes. Interstate connects directly with decentralized protocols and applications. All of your transactions happen fully onchain, and your funds remain in your custody at all times."
    }), "\n", jsx(_components.hr, {}), "\n", jsxs(_components.h2, {
      id: "bundle-checker",
      children: ["Bundle Checker", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#bundle-checker",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "A slot (or block) can process multiple buy transactions at the same time. In a bundle, several buys execute as separate transactions but land within the same block, often just milliseconds apart. While some platforms offer basic warnings, these tend to be too vague to distinguish genuine coordinated buys from normal market activity."
    }), "\n", jsx(_components.p, {
      children: "To detect bundles accurately, we apply two core rules:"
    }), "\n", jsxs(_components.h3, {
      id: "1-four-or-more-transactions-in-the-same-block",
      children: ["1. Four or more transactions in the same block", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#1-four-or-more-transactions-in-the-same-block",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "When at least four transactions occur within a single block, they get flagged as a potential bundle."
    }), "\n", jsxs(_components.h3, {
      id: "2-filtering-out-non-bundled-wallets",
      children: ["2. Filtering out non-bundled wallets", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#2-filtering-out-non-bundled-wallets",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "If a wallet participates in a detected bundle but its next transaction falls outside of a bundle, it gets excluded from detection. This removes noise from traders who aren't consistently bundling."
    }), "\n", jsx(_components.p, {
      children: "While no bundle detection method is perfect, this approach gives you a strong signal for identifying which transactions are likely part of a coordinated effort. Some false positives or missed bundles are unavoidable."
    }), "\n", jsxs(_components.h3, {
      id: "how-interstates-bundle-detection-stands-apart",
      children: ["How Interstate's bundle detection stands apart", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#how-interstates-bundle-detection-stands-apart",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Other tools like Trench Radar take a different approach, and there are important distinctions:"
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Unrestricted detection window"
      }), " — Interstate identifies bundles whenever they happen, with no time limit. Other tools often rely on flags that only catch transactions in the first few blocks after a token launches. Bundles can occur at any point, not just at creation, and they don't have to originate from the deployer or team."]
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Smarter filtering"
      }), " — We cut down on false positives by excluding wallets that don't show a consistent bundling pattern. While our bundled percentage may look different from other tools, it provides a more accurate picture of the actual risk involved in trading a specific token."]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
