// vocs/_lib/vite/.vocs/dist/assets/index-BL4gkopl.js
import { jsx, jsxs } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "layout": "landing"
};
function _createMdxContent(props) {
  const _components = {
    p: "p",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs("div", {
    style: {
      background: "linear-gradient(135deg, #0a0a0f 0%, #111827 50%, #0a0a0f 100%)",
      color: "#e5e7eb",
      minHeight: "100vh"
    },
    children: [jsxs("div", {
      style: {
        maxWidth: "1100px",
        margin: "0 auto",
        padding: "80px 24px 60px",
        textAlign: "center"
      },
      children: [jsx("img", {
        src: "/hero-banner.png",
        alt: "Interstate",
        style: {
          maxWidth: "100%",
          width: "600px",
          marginBottom: "24px"
        }
      }), jsx("p", {
        style: {
          fontSize: "clamp(18px, 2.5vw, 22px)",
          lineHeight: 1.7,
          color: "#9ca3af",
          maxWidth: "720px",
          margin: "0 auto 40px"
        },
        children: jsx(_components.p, {
          children: "The fastest and most feature-rich hybrid web trading experience, designed to elevate your crypto journey with advanced analytics and high-speed execution."
        })
      }), jsxs("div", {
        style: {
          display: "flex",
          gap: "16px",
          justifyContent: "center",
          flexWrap: "wrap"
        },
        children: [jsx("a", {
          href: "#",
          style: {
            display: "inline-flex",
            alignItems: "center",
            padding: "14px 32px",
            borderRadius: "12px",
            background: "linear-gradient(135deg, #6366f1, #4f46e5)",
            color: "#fff",
            fontSize: "16px",
            fontWeight: 600,
            textDecoration: "none",
            transition: "transform 0.2s"
          },
          children: jsx(_components.p, {
            children: "Launch App"
          })
        }), jsx("a", {
          href: "/terms-of-service",
          style: {
            display: "inline-flex",
            alignItems: "center",
            padding: "14px 32px",
            borderRadius: "12px",
            border: "1px solid rgba(99, 102, 241, 0.3)",
            background: "rgba(99, 102, 241, 0.05)",
            color: "#c7d2fe",
            fontSize: "16px",
            fontWeight: 600,
            textDecoration: "none"
          },
          children: jsx(_components.p, {
            children: "Read Docs"
          })
        })]
      })]
    }), jsx("div", {
      style: {
        maxWidth: "1100px",
        margin: "0 auto",
        padding: "40px 24px 80px",
        textAlign: "center"
      },
      children: jsx("p", {
        style: {
          fontSize: "18px",
          lineHeight: 1.8,
          color: "#9ca3af",
          maxWidth: "800px",
          margin: "0 auto"
        },
        children: jsx(_components.p, {
          children: "Buy and sell in one click, giving you the competitive edge needed in fast-paced markets. With wallet tracking and twitter monitoring, Interstate is packed with powerful features tailored for both beginners and pro traders."
        })
      })
    }), jsxs("div", {
      style: {
        maxWidth: "1100px",
        margin: "0 auto",
        padding: "0 24px 80px"
      },
      children: [jsxs("div", {
        style: {
          textAlign: "center",
          marginBottom: "48px"
        },
        children: [jsx("h2", {
          style: {
            fontSize: "36px",
            fontWeight: 700,
            color: "#f9fafb",
            margin: "0 0 16px"
          },
          children: jsx(_components.p, {
            children: "Powerful Features"
          })
        }), jsx("p", {
          style: {
            fontSize: "18px",
            color: "#6b7280",
            maxWidth: "600px",
            margin: "0 auto"
          },
          children: jsx(_components.p, {
            children: "Everything you need to trade with speed and confidence"
          })
        })]
      }), jsxs("div", {
        style: {
          display: "grid",
          gridTemplateColumns: "repeat(auto-fit, minmax(240px, 1fr))",
          gap: "20px"
        },
        children: [jsxs("div", {
          style: {
            padding: "28px",
            borderRadius: "16px",
            border: "1px solid rgba(99, 102, 241, 0.15)",
            background: "rgba(17, 24, 39, 0.6)",
            backdropFilter: "blur(8px)"
          },
          children: [jsx("div", {
            style: {
              fontSize: "28px",
              marginBottom: "12px"
            },
            children: "⚡"
          }), jsx("h3", {
            style: {
              fontSize: "18px",
              fontWeight: 600,
              color: "#f9fafb",
              margin: "0 0 8px"
            },
            children: "Buy on Migration"
          }), jsx("p", {
            style: {
              fontSize: "14px",
              color: "#6b7280",
              margin: 0,
              lineHeight: 1.6
            },
            children: "Automatically buy tokens the moment they migrate, securing early entries on new launches."
          })]
        }), jsxs("div", {
          style: {
            padding: "28px",
            borderRadius: "16px",
            border: "1px solid rgba(99, 102, 241, 0.15)",
            background: "rgba(17, 24, 39, 0.6)",
            backdropFilter: "blur(8px)"
          },
          children: [jsx("div", {
            style: {
              fontSize: "28px",
              marginBottom: "12px"
            },
            children: "📈"
          }), jsx("h3", {
            style: {
              fontSize: "18px",
              fontWeight: 600,
              color: "#f9fafb",
              margin: "0 0 8px"
            },
            children: "Sell on Migration"
          }), jsx("p", {
            style: {
              fontSize: "14px",
              color: "#6b7280",
              margin: 0,
              lineHeight: 1.6
            },
            children: "Lock in profits automatically when tokens migrate, ensuring you never miss a profitable exit."
          })]
        }), jsxs("div", {
          style: {
            padding: "28px",
            borderRadius: "16px",
            border: "1px solid rgba(99, 102, 241, 0.15)",
            background: "rgba(17, 24, 39, 0.6)",
            backdropFilter: "blur(8px)"
          },
          children: [jsx("div", {
            style: {
              fontSize: "28px",
              marginBottom: "12px"
            },
            children: "🎯"
          }), jsx("h3", {
            style: {
              fontSize: "18px",
              fontWeight: 600,
              color: "#f9fafb",
              margin: "0 0 8px"
            },
            children: "Spot Buys"
          }), jsx("p", {
            style: {
              fontSize: "14px",
              color: "#6b7280",
              margin: 0,
              lineHeight: 1.6
            },
            children: "Execute instant spot purchases with optimized routing for the best available price."
          })]
        }), jsxs("div", {
          style: {
            padding: "28px",
            borderRadius: "16px",
            border: "1px solid rgba(99, 102, 241, 0.15)",
            background: "rgba(17, 24, 39, 0.6)",
            backdropFilter: "blur(8px)"
          },
          children: [jsx("div", {
            style: {
              fontSize: "28px",
              marginBottom: "12px"
            },
            children: "💰"
          }), jsx("h3", {
            style: {
              fontSize: "18px",
              fontWeight: 600,
              color: "#f9fafb",
              margin: "0 0 8px"
            },
            children: "Quick Sell"
          }), jsx("p", {
            style: {
              fontSize: "14px",
              color: "#6b7280",
              margin: 0,
              lineHeight: 1.6
            },
            children: "One-click selling lets you exit positions instantly when speed matters most."
          })]
        }), jsxs("div", {
          style: {
            padding: "28px",
            borderRadius: "16px",
            border: "1px solid rgba(99, 102, 241, 0.15)",
            background: "rgba(17, 24, 39, 0.6)",
            backdropFilter: "blur(8px)"
          },
          children: [jsx("div", {
            style: {
              fontSize: "28px",
              marginBottom: "12px"
            },
            children: "🐦"
          }), jsx("h3", {
            style: {
              fontSize: "18px",
              fontWeight: 600,
              color: "#f9fafb",
              margin: "0 0 8px"
            },
            children: "Twitter Monitor"
          }), jsx("p", {
            style: {
              fontSize: "14px",
              color: "#6b7280",
              margin: 0,
              lineHeight: 1.6
            },
            children: "Track influential accounts in real time and act on market-moving tweets before the crowd."
          })]
        }), jsxs("div", {
          style: {
            padding: "28px",
            borderRadius: "16px",
            border: "1px solid rgba(99, 102, 241, 0.15)",
            background: "rgba(17, 24, 39, 0.6)",
            backdropFilter: "blur(8px)"
          },
          children: [jsx("div", {
            style: {
              fontSize: "28px",
              marginBottom: "12px"
            },
            children: "👁"
          }), jsx("h3", {
            style: {
              fontSize: "18px",
              fontWeight: 600,
              color: "#f9fafb",
              margin: "0 0 8px"
            },
            children: "Wallet Tracking"
          }), jsx("p", {
            style: {
              fontSize: "14px",
              color: "#6b7280",
              margin: 0,
              lineHeight: 1.6
            },
            children: "Follow top-performing wallets and mirror successful traders' moves in real time."
          })]
        }), jsxs("div", {
          style: {
            padding: "28px",
            borderRadius: "16px",
            border: "1px solid rgba(99, 102, 241, 0.15)",
            background: "rgba(17, 24, 39, 0.6)",
            backdropFilter: "blur(8px)"
          },
          children: [jsx("div", {
            style: {
              fontSize: "28px",
              marginBottom: "12px"
            },
            children: "📊"
          }), jsx("h3", {
            style: {
              fontSize: "18px",
              fontWeight: 600,
              color: "#f9fafb",
              margin: "0 0 8px"
            },
            children: "Limit Orders"
          }), jsx("p", {
            style: {
              fontSize: "14px",
              color: "#6b7280",
              margin: 0,
              lineHeight: 1.6
            },
            children: "Set precise entry and exit points with advanced limit orders that execute automatically."
          })]
        }), jsxs("div", {
          style: {
            padding: "28px",
            borderRadius: "16px",
            border: "1px solid rgba(99, 102, 241, 0.15)",
            background: "rgba(17, 24, 39, 0.6)",
            backdropFilter: "blur(8px)"
          },
          children: [jsx("div", {
            style: {
              fontSize: "28px",
              marginBottom: "12px"
            },
            children: "🔧"
          }), jsx("h3", {
            style: {
              fontSize: "18px",
              fontWeight: 600,
              color: "#f9fafb",
              margin: "0 0 8px"
            },
            children: "Migration Tools"
          }), jsx("p", {
            style: {
              fontSize: "14px",
              color: "#6b7280",
              margin: 0,
              lineHeight: 1.6
            },
            children: "Complete suite of migration tools to help you capitalize on every token launch event."
          })]
        })]
      })]
    }), jsx("div", {
      style: {
        maxWidth: "1100px",
        margin: "0 auto",
        padding: "0 24px 80px"
      },
      children: jsxs("div", {
        style: {
          padding: "48px",
          borderRadius: "20px",
          border: "1px solid rgba(99, 102, 241, 0.2)",
          background: "linear-gradient(135deg, rgba(99, 102, 241, 0.08) 0%, rgba(17, 24, 39, 0.6) 100%)",
          textAlign: "center"
        },
        children: [jsx("h2", {
          style: {
            fontSize: "32px",
            fontWeight: 700,
            color: "#f9fafb",
            margin: "0 0 16px"
          },
          children: jsx(_components.p, {
            children: "Every Second Counts"
          })
        }), jsx("p", {
          style: {
            fontSize: "18px",
            lineHeight: 1.8,
            color: "#9ca3af",
            maxWidth: "700px",
            margin: "0 auto"
          },
          children: jsx(_components.p, {
            children: "In the market of token launches and high-frequency trading, every second counts, and Interstate is your ultimate tool for securing massive gains. Whether you're here to capitalize on the latest opportunities or execute trades with confidence, Interstate provides you with precision, speed, and reliability."
          })
        })]
      })
    }), jsxs("div", {
      style: {
        maxWidth: "1100px",
        margin: "0 auto",
        padding: "0 24px 80px",
        textAlign: "center"
      },
      children: [jsx("h2", {
        style: {
          fontSize: "36px",
          fontWeight: 700,
          color: "#f9fafb",
          margin: "0 0 16px"
        },
        children: jsx(_components.p, {
          children: "Our Mission"
        })
      }), jsx("p", {
        style: {
          fontSize: "18px",
          lineHeight: 1.8,
          color: "#9ca3af",
          maxWidth: "800px",
          margin: "0 auto 40px"
        },
        children: jsx(_components.p, {
          children: "At Interstate, our mission is to build a comprehensive, all-in-one trading platform that not only delivers unmatched performance but also rewards our users for being part of our growth. We are committed to being the fastest, most efficient, and most community-driven trading experience in the market—implementing the features you want most while ensuring a seamless and rewarding journey."
        })
      }), jsx("p", {
        style: {
          fontSize: "16px",
          fontWeight: 600,
          color: "#818cf8"
        },
        children: jsx(_components.p, {
          children: "— The Interstate Team"
        })
      })]
    }), jsxs("div", {
      style: {
        maxWidth: "1100px",
        margin: "0 auto",
        padding: "0 24px 100px",
        textAlign: "center"
      },
      children: [jsx("h2", {
        style: {
          fontSize: "32px",
          fontWeight: 700,
          color: "#f9fafb",
          margin: "0 0 16px"
        },
        children: jsx(_components.p, {
          children: "Ready to Trade?"
        })
      }), jsx("p", {
        style: {
          fontSize: "18px",
          color: "#9ca3af",
          margin: "0 0 32px"
        },
        children: jsx(_components.p, {
          children: "Join Interstate and experience the future of trading today."
        })
      }), jsx("a", {
        href: "#",
        style: {
          display: "inline-flex",
          alignItems: "center",
          padding: "16px 40px",
          borderRadius: "12px",
          background: "linear-gradient(135deg, #6366f1, #4f46e5)",
          color: "#fff",
          fontSize: "18px",
          fontWeight: 600,
          textDecoration: "none"
        },
        children: jsx(_components.p, {
          children: "Get Started"
        })
      })]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
