// vocs/_lib/vite/.vocs/dist/assets/insurance-CAycfZ6o.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Insurance",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    header: "header",
    img: "img",
    p: "p",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "insurance",
        children: ["Insurance", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#insurance",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.p, {
      children: ["Interstate is currently in whitelisted mode, only whitelisted users can send a mainnet preconfirmation request. In order to join the whitelist please ping us on telegram:  ", jsx(_components.a, {
        href: "%5Bhttps://t.me/+Pcs9bykxK3BiMzk5%5D(https://t.me/+-i4dP7U2BggxMzAx)",
        children: jsx(_components.img, {
          src: "https://img.shields.io/endpoint?color=neon&logo=telegram&label=chat&url=https%3A%2F%2Ftg.sumanjay.workers.dev%2F%2BPcs9bykxK3BiMzk5",
          alt: "Chat"
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Once interstate enables general mainnet, we will have two insurance guarantees, proposers can opt into the one of their choosing and preconfirmation requestors can decide whether to send a request or not based on if the insurance type and quantity offered by each proposer is sufficient."
    }), "\n", jsx(_components.p, {
      children: "The two insurance types will be native eth and restaked eth. Native eth means the preconfirmation requestor will be compensated in native eth. Restaked eth means the requestor will be compensated in restaked eth by the slashing policies defined by the restaking networks: Karak, Eigenlayer, Symbiotic."
    }), "\n", jsxs(_components.p, {
      children: ["Native eth contracts are here: ", jsx(_components.a, {
        href: "https://github.com/eth-fabric/urc",
        children: "https://github.com/eth-fabric/urc"
      }), "."]
    }), "\n", jsxs(_components.p, {
      children: ["Restaked eth contracts are here: ", jsx(_components.a, {
        href: "https://github.com/interstate-labs/restaking-urc",
        children: "https://github.com/interstate-labs/restaking-urc"
      })]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
