// vocs/_lib/vite/.vocs/dist/assets/intro-mZnwv-4y.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Intro",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    h2: "h2",
    header: "header",
    li: "li",
    p: "p",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "intro",
        children: ["Intro", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#intro",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Interstate currently offers:"
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Inclusion Preconfirmations"
      }), "\n", jsx(_components.li, {
        children: "Execution Preconfirmations"
      }), "\n", jsx(_components.li, {
        children: "Other Proposer Commitments"
      }), "\n"]
    }), "\n", jsxs(_components.h2, {
      id: "inclusion-preconfirmation",
      children: ["Inclusion Preconfirmation", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#inclusion-preconfirmation",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["A preconfirmation is a sale of one or more transaction in an upcoming ethereum block prior to the block occuring. The proposer commitment is backed by restaking netwoks like ", jsx(_components.a, {
        href: "https://karak.network",
        children: "Karak"
      }), " and ", jsx(_components.a, {
        href: "https://eigenlayer.xyz",
        children: "Eigenlayer"
      }), ". The commitment is stake backed, and slashed stake will be used to compensate the requester if their preconfirmation is not honored."]
    }), "\n", jsx(_components.p, {
      children: "Space in the block can be bought 1 - ~32 slots in advance (the rough size of the ethereum proposer lookahead). We expect 90% of validators to run a preconfirmation sidecar (same as 90% validators today run flashbots) and offer preconfirmations."
    }), "\n", jsx(_components.p, {
      children: "Pricing: The preconfirmation buyer can expect to pay 1-10% of gas in addition gas fees to buy an inclusion preconfirmation."
    }), "\n", jsx(_components.p, {
      children: "Inclusion preconfirmations give a weaker guarantee than execution preconfirmations in that the transaction may revert if the gas limit of the block is exceeded, or if there is a revert on the transction."
    }), "\n", jsx(_components.p, {
      children: "So if you buy an inclusion preconfirmation, your transaction will be included in the block but may revert. Execution preconfirmations are guaranteed to not revert."
    }), "\n", jsxs(_components.h2, {
      id: "execution-preconfirmation-coming-soon",
      children: ["Execution Preconfirmation (coming soon)", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#execution-preconfirmation-coming-soon",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "An execution preconfirmation is guaranteed to be included in the upcoming block and guarantees the transaction will not revert. Essentially the validator is selling the txs in the block one by one, with the top of the block being sold first."
    }), "\n", jsx(_components.p, {
      children: "This gives the preconfirmation buyer the guarantee that their transaction will not revert."
    }), "\n", jsxs(_components.h2, {
      id: "other-proposer-commitments-coming-soon",
      children: ["Other Proposer Commitments (coming soon)", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#other-proposer-commitments-coming-soon",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "These allow the proposer to commit to generic attestations."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
