// vocs/_lib/vite/.vocs/dist/assets/long-form-setup-C_Igm4CN.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Running Your Own RPC",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    header: "header",
    img: "img",
    p: "p",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "running-your-own-rpc",
        children: ["Running Your Own RPC", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#running-your-own-rpc",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.p, {
      children: ["Run your own RPC by building and deploying the docker image provided here: ", jsx(_components.a, {
        href: "https://github.com/interstate-labs/proposer-commitment-network/tree/main/interstate-gateway",
        children: "https://github.com/interstate-labs/proposer-commitment-network/tree/main/interstate-gateway"
      })]
    }), "\n", jsxs(_components.p, {
      children: ["We also provide help charts to do so here: ", jsx(_components.a, {
        href: "https://github.com/interstate-labs/proposer-commitment-network/tree/main/kubernetes",
        children: "https://github.com/interstate-labs/proposer-commitment-network/tree/main/kubernetes"
      })]
    }), "\n", jsxs(_components.p, {
      children: ["You need to set the .env file, and can also run from cargo source. When pointing relays ensure the relays you point to follow the constraints api: ", jsx(_components.a, {
        href: "https://github.com/eth-fabric/constraints-specs",
        children: "https://github.com/eth-fabric/constraints-specs"
      }), "."]
    }), "\n", jsxs(_components.p, {
      children: ["We suggest pinging us on on telegram for guided help: ", jsx(_components.a, {
        href: "%5Bhttps://t.me/+Pcs9bykxK3BiMzk5%5D(https://t.me/+-i4dP7U2BggxMzAx)",
        children: jsx(_components.img, {
          src: "https://img.shields.io/endpoint?color=neon&logo=telegram&label=chat&url=https%3A%2F%2Ftg.sumanjay.workers.dev%2F%2BPcs9bykxK3BiMzk5",
          alt: "Chat"
        })
      })]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
