// vocs/_lib/vite/.vocs/dist/assets/quickstart-CSaseVAW.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Send A Preconf Request",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    header: "header",
    img: "img",
    p: "p",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "send-a-preconf-request",
        children: ["Send A Preconf Request", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#send-a-preconf-request",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Endpoint"
          }), jsx(_components.th, {
            children: "Helder"
          }), jsx(_components.th, {
            children: "Mainnet"
          }), jsx(_components.th, {
            children: "Hoodi"
          })]
        })
      }), jsx(_components.tbody, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "rpc-endpoint"
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://helder-rpc.interstate.so",
              children: "https://helder-rpc.interstate.so"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://rpc.interstate.so",
              children: "https://rpc.interstate.so"
            })
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://hoodi-rpc.interstate.so",
              children: "https://hoodi-rpc.interstate.so"
            })
          })]
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["You should point to the rpc endpoint of the respective network.\nYou can send a preconfirmation request quickly by setting the env file and running this: ", jsx(_components.a, {
        href: "https://github.com/interstate-labs/preconf-cli",
        children: "https://github.com/interstate-labs/preconf-cli"
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Be sure to ping the Interstate team on telegram ", jsx(_components.a, {
        href: "%5Bhttps://t.me/+Pcs9bykxK3BiMzk5%5D(https://t.me/+-i4dP7U2BggxMzAx)",
        children: jsx(_components.img, {
          src: "https://img.shields.io/endpoint?color=neon&logo=telegram&label=chat&url=https%3A%2F%2Ftg.sumanjay.workers.dev%2F%2BPcs9bykxK3BiMzk5",
          alt: "Chat"
        })
      }), " for guided setup help from us and to request testnet eth!"]
    }), "\n", jsx(_components.p, {
      children: "Interstate is currently in whitelisted mode, only whitelisted users can send a mainnet preconfirmation request. In order to join the whitelist please ping us on telegram"
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
