// vocs/_lib/vite/.vocs/dist/assets/quickstart-ByFf5gow.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Validator Quickstart",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    em: "em",
    h1: "h1",
    header: "header",
    img: "img",
    li: "li",
    ol: "ol",
    p: "p",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "validator-quickstart",
        children: ["Validator Quickstart", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#validator-quickstart",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "start-interstate-on-your-validator-and-earn-rewards",
        children: ["Start Interstate On Your Validator and Earn Rewards", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#start-interstate-on-your-validator-and-earn-rewards",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.p, {
      children: ["Be sure to ping the Interstate team on ", jsx(_components.a, {
        href: "https://t.me/+-i4dP7U2BggxMzAx",
        children: "telegram"
      }), " ", jsx(_components.img, {
        src: "https://img.shields.io/endpoint?color=neon&logo=telegram&label=chat&url=https%3A%2F%2Ftg.sumanjay.workers.dev%2F%2BPcs9bykxK3BiMzk5",
        alt: "Chat"
      }), " for guided setup help from us and to request testnet eth!"]
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Description"
          }), jsx(_components.th, {
            children: "Endpoint"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "helder-endpoint"
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://0x8134fe68477ecf1d5402868504f50303be599c916a45568a1bbe60dafec94b37a5650d5ca9589a9a69f126cde80e8edb@helder-builder-proxy.interstate.so:443",
              children: "https://0x8134fe68477ecf1d5402868504f50303be599c916a45568a1bbe60dafec94b37a5650d5ca9589a9a69f126cde80e8edb@helder-builder-proxy.interstate.so:443"
            })
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "mainnet-endpoint"
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://0x9110847c15a7f5c80a9fdd5db989a614cc01104e53bd8c252b6f46a4842c7fdef6b9593336035b5094878deff386804c@builder-proxy.interstate.so:443",
              children: "https://0x9110847c15a7f5c80a9fdd5db989a614cc01104e53bd8c252b6f46a4842c7fdef6b9593336035b5094878deff386804c@builder-proxy.interstate.so:443"
            })
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "hoodi-endpoint"
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://0x9110847c15a7f5c80a9fdd5db989a614cc01104e53bd8c252b6f46a4842c7fdef6b9593336035b5094878deff386804c@hoodi-builder-proxy-alpha.interstate.so:443",
              children: "https://0x9110847c15a7f5c80a9fdd5db989a614cc01104e53bd8c252b6f46a4842c7fdef6b9593336035b5094878deff386804c@hoodi-builder-proxy-alpha.interstate.so:443"
            })
          })]
        })]
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Set your commit-boost or mev-boost relay to Interstate's endpoint. Set Interstate as the ", jsx(_components.em, {
        children: "only"
      }), " relay."]
    }), "\n", jsxs(_components.ol, {
      children: ["\n", jsxs(_components.li, {
        children: ["Point MEV-Boost or Commit-Boost to the relay endpoint above. Make sure you set only one relay endpoint, the relay above. Interstate is an rpc that submits constraints to relays.", "\n", jsxs(_components.ul, {
          children: ["\n", jsx(_components.li, {
            children: "Your request follows this path: commit/mev-boost --> rpc --> relay"
          }), "\n", jsxs(_components.li, {
            children: ["You can run the RPC yourself if you would like to configure which relays to use. Any ", jsx(_components.a, {
              href: "https://github.com/eth-fabric/constraints-specs",
              children: "constraints-api"
            }), " compatible relay works."]
          }), "\n"]
        }), "\n"]
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: "List of Relays that Interstate uses:"
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Network"
          }), jsx(_components.th, {
            children: "Constraints-Api-Compatible-Relay-Endpoint"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "Helder"
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://helder-relay.interstate.so",
              children: "https://helder-relay.interstate.so"
            })
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "Mainnet"
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://mainnet-relay.interstate.so",
              children: "https://mainnet-relay.interstate.so"
            })
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "Mainnet"
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://bloxroute.max-profit.blxrbdn.com",
              children: "https://bloxroute.max-profit.blxrbdn.com"
            })
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: "Hoodi"
          }), jsx(_components.td, {
            children: jsx(_components.a, {
              href: "https://hoodi-relay.interstate.so",
              children: "https://hoodi-relay.interstate.so"
            })
          })]
        })]
      })]
    }), "\n", jsx(_components.p, {
      children: "Coming soon: Aestus, Ultrasound Relay, Titan-Relay, and more."
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: ["Quickly spin up a helder validator by running: ", jsx(_components.a, {
          href: "https://github.com/interstate-labs/helder-validator-quickstart",
          children: "https://github.com/interstate-labs/helder-validator-quickstart"
        })]
      }), "\n"]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
