// vocs/_lib/vite/.vocs/dist/assets/rewards-BBIsgsi2.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "The Outpost",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    hr: "hr",
    li: "li",
    p: "p",
    strong: "strong",
    table: "table",
    tbody: "tbody",
    td: "td",
    th: "th",
    thead: "thead",
    tr: "tr",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "the-outpost",
        children: ["The Outpost", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#the-outpost",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsxs(_components.h2, {
      id: "ranks--rewards",
      children: ["Ranks & Rewards", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#ranks--rewards",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Claim up to ", jsx(_components.strong, {
        children: "45% cashback"
      }), " on every trade."]
    }), "\n", jsxs(_components.p, {
      children: ["Head over to ", jsx(_components.a, {
        href: "https://interstate.so",
        children: "Interstate.so"
      }), " and complete the required steps to start earning your cashback right away."]
    }), "\n", jsx(_components.hr, {}), "\n", jsxs(_components.h2, {
      id: "rewards",
      children: ["Rewards", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#rewards",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.h3, {
      id: "gold",
      children: ["Gold", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#gold",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Gold is the foundation of the Outpost reward system. You accumulate Gold with every action you take on the platform — every trade executed and every quest completed adds to your balance."
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Trade to earn"
        }), " — Receive Gold for every dollar in volume you generate."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Daily streaks"
        }), " — Keep your trading streak alive for daily Gold bonuses. The longer your streak, the larger the reward."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Rank progression"
        }), " — Climbing through the ranks earns you Gold payouts and unlocks higher Gold multipliers."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Quests"
        }), " — Complete daily and seasonal quests to collect additional Gold."]
      }), "\n", jsxs(_components.li, {
        children: [jsx(_components.strong, {
          children: "Special events"
        }), " — Watch for limited-time quests that award bonus Gold and multiplier boosts."]
      }), "\n"]
    }), "\n", jsxs(_components.h3, {
      id: "solana",
      children: ["Solana", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#solana",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Solana rewards come from two sources: Cashback and the Referral System. As you advance through the Outpost ranks and engage with quests, your cashback can reach up to ", jsx(_components.strong, {
        children: "45%"
      }), ", and your referral fee share can climb as high as ", jsx(_components.strong, {
        children: "50%"
      }), "."]
    }), "\n", jsx(_components.hr, {}), "\n", jsxs(_components.h2, {
      id: "outpost-ranks",
      children: ["Outpost Ranks", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#outpost-ranks",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "The Outpost is built around progression. Each rank you reach unlocks larger multipliers, better rewards, and exclusive perks."
    }), "\n", jsxs(_components.table, {
      children: [jsx(_components.thead, {
        children: jsxs(_components.tr, {
          children: [jsx(_components.th, {
            children: "Rank"
          }), jsx(_components.th, {
            children: "Cashback"
          }), jsx(_components.th, {
            children: "Gold Multiplier"
          })]
        })
      }), jsxs(_components.tbody, {
        children: [jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Degen"
            })
          }), jsx(_components.td, {
            children: "10%"
          }), jsx(_components.td, {
            children: "1x"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Warrior"
            })
          }), jsx(_components.td, {
            children: "15%"
          }), jsx(_components.td, {
            children: "2x"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Gladiator"
            })
          }), jsx(_components.td, {
            children: "20%"
          }), jsx(_components.td, {
            children: "2.5x"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Commander"
            })
          }), jsx(_components.td, {
            children: "25%"
          }), jsx(_components.td, {
            children: "3x"
          })]
        }), jsxs(_components.tr, {
          children: [jsx(_components.td, {
            children: jsx(_components.strong, {
              children: "Titan"
            })
          }), jsx(_components.td, {
            children: "35%"
          }), jsx(_components.td, {
            children: "4x"
          })]
        })]
      })]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
