// vocs/_lib/vite/.vocs/dist/assets/slashing_insurance_registration-h_WsICDI.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Register For Slashing Insurance",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    header: "header",
    p: "p",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "register-for-slashing-insurance",
        children: ["Register For Slashing Insurance", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#register-for-slashing-insurance",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Each validator can register for slashing insurance at their discretion. The steps for the validator to register for slashing insurance are here:"
    }), "\n", jsx(_components.p, {
      children: jsx(_components.a, {
        href: "https://github.com/interstate-labs/proposer-commitment-network-dev/blob/main/scripts/urc/README.md",
        children: "https://github.com/interstate-labs/proposer-commitment-network-dev/blob/main/scripts/urc/README.md"
      })
    }), "\n", jsx(_components.p, {
      children: "Registering for slashing insurance allows users to slash if their preconfirmation is missed and increases the chances of a user signing up for preconfirmations. Certain users may only send preconfirmations if the slash insurance for a certain slot exceeds a certain amount. Interstate can provide slashing collateral on your behalf, please reach out to inquire!"
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
