// vocs/_lib/vite/.vocs/dist/assets/support-O5pssF1I.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Support",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    aside: "aside",
    div: "div",
    h1: "h1",
    h2: "h2",
    header: "header",
    hr: "hr",
    p: "p",
    strong: "strong",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "support",
        children: ["Support", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#support",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "This page provides clear instructions to join our Telegram group and get help from the team. Follow each step carefully to avoid scams and ensure a smooth experience."
    }), "\n", jsxs(_components.p, {
      children: [jsx(_components.strong, {
        children: "Join our Telegram:"
      }), " ", jsx(_components.a, {
        href: "https://t.me/+DDXGrsJoe3szYTAx",
        children: "Join Here"
      })]
    }), "\n", jsx(_components.aside, {
      "data-callout": "warning",
      children: jsx(_components.p, {
        children: "In trading groups, scammers often impersonate support staff or offer unsolicited help to gain your trust. Always disregard direct messages from individuals claiming to be support. Only seek assistance by tagging a moderator in the group."
      })
    }), "\n", jsx(_components.hr, {}), "\n", jsxs(_components.h2, {
      id: "join-our-telegram",
      children: ["Join Our Telegram", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#join-our-telegram",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "To join our Telegram group, click the link found above. If you do not have a Telegram account yet, download the app and create one, then return to the link to join the group."
    }), "\n", jsx(_components.hr, {}), "\n", jsxs(_components.h2, {
      id: "ignore-any-direct-messages",
      children: ["Ignore Any Direct Messages", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#ignore-any-direct-messages",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "Once you've joined, make sure to ignore any direct messages from people claiming to be support or offering help. Official assistance will only be provided within the group by moderators."
    }), "\n", jsx(_components.hr, {}), "\n", jsxs(_components.h2, {
      id: "getting-help",
      children: ["Getting Help", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#getting-help",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "If you need assistance, tag a moderator directly in the group chat. Describe your issue clearly and a member of our team will respond as soon as possible."
    }), "\n", jsx(_components.hr, {}), "\n", jsxs(_components.h2, {
      id: "providing-information--waiting",
      children: ["Providing Information & Waiting", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#providing-information--waiting",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsx(_components.p, {
      children: "When asking for help, make sure to provide detailed and accurate information about your issue. The more in-depth your explanation, the faster our staff can help resolve your question. Remember, our staff are working very hard to assist everyone, so please be respectful and patient while waiting for a response."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
