// vocs/_lib/vite/.vocs/dist/assets/terms-of-service-CwKcpmJJ.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
const frontmatter = {
  "title": "Terms of Service",
  "description": "undefined"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    div: "div",
    h1: "h1",
    header: "header",
    p: "p",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsx(_components.header, {
      children: jsxs(_components.h1, {
        id: "terms-of-service",
        children: ["Terms of Service", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#terms-of-service",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      })
    }), "\n", jsx(_components.p, {
      children: "Please see our terms of service here:"
    }), "\n", jsx(_components.p, {
      children: jsx(_components.a, {
        href: "https://drive.google.com/file/d/1x4VAC6RW-OJVEHx9NvGNqnwh2oPyicAo/view?usp=sharing",
        children: "https://drive.google.com/file/d/1x4VAC6RW-OJVEHx9NvGNqnwh2oPyicAo/view?usp=sharing"
      })
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
